docs/guide/controller-programming.md

```diff
@@ -431,24 +1314,93 @@ In this case the controller should just save the experiment results and quit by
 This will terminate the controller process and freeze the simulation at the current simulation step.
 The physics simulation and every robot involved in the simulation will stop.
 
+%tab-component "language"
+%tab "C"
 ```c
+#include <stdlib.h>
+
 // freeze the whole simulation
 if (finished) {
   saveExperimentData();
   exit(0);
 }
 ```
+%tab-end
+
+%tab "C++"
+```cpp
+#include <cstdlib>
+
+// freeze the whole simulation
+if (finished) {
+  saveExperimentData();
+  exit(0);
+}
+```
+%tab-end
+
+%tab "Python"
+```python
+import sys
+
+# freeze the whole simulation
+if finished:
+  saveExperimentData()
+  sys.exit(0)
+```
+%tab-end
+
+%tab "Java"
+```java
+// freeze the whole simulation
+if (finished) {
+  saveExperimentData();
+  System.exit(0);
+}
+```
+%tab-end
 
+%tab "MATLAB"
+```MATLAB
+% freeze the whole simulation
+if finished
+  saveExperimentData();
+  quit(0);
+end
+```
+%tab-end
+%end
+
+Please note that this specific use case is only possible in C/C++ since controllers written in a different language will handle the cleanup part their own way.
 If only one robot controller needs to terminate but the simulation should continue with the other robots, then the terminating robot should call the `wb_robot_cleanup` function right before quitting:
 
+%tab-component "language"
+%tab "C"
 ```c
+#include <stdlib.h>
+
 // terminate only this robot controller
 if (finished) {
-  saveExperimentsData();
+  saveExperimentData();
   wb_robot_cleanup();
   exit(0);
 }
 ```
+%tab-end
+
+%tab "C++"
+```cpp
+#include <cstdlib>
+
+// terminate only this robot controller
+if (finished) {
+  saveExperimentData();
+  delete robot;
+  exit(0);
+}
+```
+%tab-end
+%end
```

Comment:
I would remove all this and move the `wb_robot_cleanup()`  and `delete robot;` statements in the above code snippet.